base/base.py

```diff
@@ -17,8 +21,51 @@ def read_app_data(self, file_name, file_path=None):
         	Returns:
         	    Parsed data in dictionary
         '''
-        if td_dir is None:
-            td_dir = self.test["case_td_dir"]
+        if file_name is None or class_name is None:
+            print('Cannot get the test data file name') 
+            raise Exception
+        else:
+            stack = inspect.stack()[1]
+            if dir_name is None:
+                dir_name = os.path.basename(stack[1])
+            if fn_name is None:
+                fn_name = stack[3]          
+            file = os.getcwd() + '/applications/sample/testdata' + '/' + dir_name + '/' + class_name +'/' + fn_name + '/' + file_name
```

Comment:
Why hardcoding "applications/sample/testdata"
We need to use this method from different projects.